pym/bob/cmds/build.py

```diff
@@ -961,6 +984,29 @@ def commonBuildDevelop(parser, argv, bobRoot, develop):
     recipes.setConfigFiles(args.configFile)
     recipes.parse()
 
+    if not args.ignore_userconfig:
+        if develop:
+            cfg = recipes.getUserConfig().get('dev', {})
+        else:
+            cfg = recipes.getUserConfig().get('build', {})
+        args.destination = cfg.get('destination', args.destination)
+        args.force = cfg.get('force', args.force)
+        args.no_deps = cfg.get('no-deps', args.no_deps)
+        buildmode = cfg.get('build-mode', None)
+        if buildmode == 'build-only':
+            args.build_only = True
+            args.checkout_only = False
```

Comment:
What about defining the arguments the following way:?

```python
    group = parser.add_mutually_exclusive_group()
    group.add_argument('-b', '--build-only', dest='build_mode', default="normal",
        action='store_const', const='build-only', 
        help="Don't checkout, just build and package")
    group.add_argument('-B', '--checkout-only', dest='build_mode',
        action='store_const', const='checkout-only', 
        help="Don't build, just check out sources")
    group.add_argument('--normal', dest='build_mode',
        action='store_const', const='normal', 
        help="Checkout, build and package")
```

Then it's easy to merge it with the default settings.